onadata/apps/api/viewsets/dataview_viewset.py

```diff
@@ -77,25 +77,47 @@ def get_field_lookup(column, filter_string):
     return "json__" + column + filter_to_field_lookup(filter_string)
 
 
-def apply_filters(instance_qs, filters):
+def get_filter_kwargs(filters):
     """
     Apply filters on a queryset
     """
+    kwargs = {}
     if filters:
         for f in filters:
-            value = f['value']
+            value = f"{f['value']}"
             column = f['column']
-            instance_qs = instance_qs.filter(
-                **{
-                    get_field_lookup(column, f['filter']):
-                    value
-                }
-            )
+            filter_kwargs = {
+                get_field_lookup(column, f['filter']):
+                value
+            }
+            kwargs = {
+                **kwargs,
+                **filter_kwargs
+            }
+    return kwargs
+
+
+def apply_filters(instance_qs, filters):
+    """
+    Apply filters on a queryset
+    """
+    if filters:
+        return instance_qs.filter(**get_filter_kwargs(filters))
     return instance_qs
 
-# pylint: disable=too-many-ancestors
+
+def get_dataview_instances(dataview):
+    """
+    Get all instances that belong to ths dataview
+    """
+    return apply_filters(
+        dataview.xform.instances.filter(
+            deleted_at__isnull=True
+        ), dataview.query
+    ).order_by('id')
```

Comment:
Do we have to do the ordering ? Can we just let the instances be returned in the same order stored in the database

Reply:
No, this isn't absolutely necessary; I think the instances are already sorted by ID in the DB

Reply:
Removing